### PR TITLE
Handle return codes from OpenSSL BIO_*

### PR DESCRIPTION
**Pull request by zeha**
Fixes warning from gcc.

### CODE REVIEW

File: pdns/base64.cc [hunk 1]
```diff
@@ -58,8 +62,10 @@ std::string Base64Encode (const std::string& src)
     bio = BIO_new(BIO_s_mem());
     bio = BIO_push(b64, bio);
     BIO_set_flags(bio, BIO_FLAGS_BASE64_NO_NL);
-    BIO_write(bio, src.c_str(), src.length());
-    BIO_flush(bio);
+    if (BIO_write(bio, src.c_str(), src.length() != src.length())) {
```
Comment (rgacogne) [inline pdns/base64.cc]: I think you have misplaced a ')', here src.length() != src.length() is not going to work well :)

File: pdns/base64.cc [hunk 2]
```diff
@@ -32,6 +33,9 @@ int B64Decode(const std::string& src, std::string& dst)
   bio = BIO_push(b64, bio);
   BIO_set_flags(bio, BIO_FLAGS_BASE64_NO_NL);
   olen = BIO_read(b64, d.get(), dlen);
+  if ((olen == 0 || olen == -1) && BIO_should_retry(bio)) {
+    throw std::runtime_error("BIO_read failed to read all data from memory buffer");
```
Comment (rgacogne) [inline pdns/base64.cc]: Doesn't that leak bio?

**Comment by Habbie** [comment]
ping

**Comment by pieterlexis** [comment]
ping!

**Comment by zeha** [comment]
Rebased on top of current master to get rid of merge issues, and fixed the possible BIO leaks, plus actually compiles ;-)

**Pull request merged**